packages/dd-trace/src/ci-visibility/dynamic-instrumentation/worker/index.js

```diff
@@ -98,17 +99,21 @@ async function addBreakpoint (probe) {
     throw new Error(`No loaded script found for ${file}`)
   }
 
-  const [path, scriptId, sourceMapURL] = script
+  const { url, scriptId, sourceMapURL, source } = script
 
-  log.warn(`Adding breakpoint at ${path}:${line}`)
+  log.warn(`Adding breakpoint at ${url}:${line}`)
 
   let lineNumber = line
 
-  if (sourceMapURL && sourceMapURL.startsWith('data:')) {
+  if (sourceMapURL) {
     try {
-      lineNumber = await processScriptWithInlineSourceMap({ file, line, sourceMapURL })
+      lineNumber = await getSourceMappedLine(url, source, line, sourceMapURL)
     } catch (err) {
-      log.error('Error processing script with inline source map', err)
+      log.error('Error processing script with source map', err)
+    }
+    if (lineNumber === null) {
+      log.error('Could not find generated position for %s:%s', url, line)
+      lineNumber = line
     }
```

Comment:
Hopefully the new algorithm is more stable in regards to finding the right line. If we still can't find it, shouldn't this code just throw so it's aborted?

Reply:
I wanted to avoid throwing so that the next test may have a chance to still set a probe

Reply:
Shouldn't you then at least return early and not try to attach the breakpoint to a line number is most likely incorrect?

Reply:
I know you try-catch around the code that adds the breakpoint takes care of that for you, but if you don't think it will work, I think it makes the purpose of the code more readable if we return early in that case

Reply:
> Shouldn't you then at least return early and not try to attach the breakpoint to a line number that is most likely incorrect?

you're right. I remember finding a situation where the source map was there, `processScriptWithInlineSourceMap` failed, but setting the probe in the original line number worked 😓. I'm definitely missing something here